lib/jxl/enc_heuristics.cc

```diff
@@ -207,6 +207,121 @@ bool DefaultEncoderHeuristics::HandlesColorConversion(
+      0.010727545221453,   -0.039563875032655,  -0.0047187050820992,
+      -0.0042216781245469, -0.0012016220643441, 0.01473598935701,
+      0.024831698004347,   0.0047301162858671,  -0.031030197792468,
+      -0.031030197792468,  0.0047301162858671,  0.024831698004347,
+      0.01473598935701,    -0.0012016220643441, -0.0042216781245469,
+      0.0073616744679974,  0.083265651677212,   -0.0051874044212185,
+      -0.19568935972668,   -0.031030197792468,  0.38964539007717,
+      0.38964539007717,    -0.031030197792468,  -0.19568935972668,
+      -0.0051874044212185, 0.083265651677212,   0.0073616744679974,
+      0.0073616744679974,  0.083265651677212,   -0.0051874044212185,
+      -0.19568935972668,   -0.031030197792468,  0.38964539007717,
+      0.38964539007717,    -0.031030197792468,  -0.19568935972668,
+      -0.0051874044212185, 0.083265651677212,   0.0073616744679974,
+      -0.0042216781245469, -0.0012016220643441, 0.01473598935701,
+      0.024831698004347,   0.0047301162858671,  -0.031030197792468,
+      -0.031030197792468,  0.0047301162858671,  0.024831698004347,
+      0.01473598935701,    -0.0012016220643441, -0.0042216781245469,
+      -0.0047187050820992, -0.039563875032655,  0.010727545221453,
+      0.11040774078162,    0.024831698004347,   -0.19568935972668,
+      -0.19568935972668,   0.024831698004347,   0.11040774078162,
+      0.010727545221453,   -0.039563875032655,  -0.0047187050820992,
+      0.0024550503214722,  -0.0040179839838831, -0.0089432318159899,
+      0.010727545221453,   0.01473598935701,    -0.0051874044212185,
+      -0.0051874044212185, 0.01473598935701,    0.010727545221453,
+      -0.0089432318159899, -0.0040179839838831, 0.0024550503214722,
+      0.0019813989090779,  0.017461064645154,   -0.0040179839838831,
+      -0.039563875032655,  -0.0012016220643441, 0.083265651677212,
+      0.083265651677212,   -0.0012016220643441, -0.039563875032655,
+      -0.0040179839838831, 0.017461064645154,   0.0019813989090779,
+      0.017242556564832,   0.0019813989090779,  0.0024550503214722,
+      -0.0047187050820992, -0.0042216781245469, 0.0073616744679974,
+      0.0073616744679974,  -0.0042216781245469, -0.0047187050820992,
+      0.0024550503214722,  0.0019813989090779,  0.017242556564832};
+
+  int64_t xsize = input.xsize();
+  int64_t ysize = input.ysize();
+
+  for (size_t y = 0; y < output->ysize(); y++) {
+    float* row_out = output->Row(y);
+    const float* row_in[kernely];
+    // get the rows in the support
+    for (size_t ky = 0; ky < kernely; ky++) {
+      int64_t iy = y * 2 + ky - (kernely - 1) / 2;
+      if (iy < 0) iy = 0;
+      if (iy >= ysize) iy = ysize - 1;
+      row_in[ky] = input.Row(iy);
+    }
+
+    for (size_t x = 0; x < output->xsize(); x++) {
+      // get min and max values of the original image in the support
+      float min = 1e300;
+      float max = -1e300;
+      for (int64_t ky = 5; ky + 5 < kernely; ky++) {
```

Comment:
make 5 a named constant, that will make the code more obvious

Reply:
Done